pkg/postage/batchstore/reserve.go

```diff
@@ -122,6 +122,46 @@ func (s *store) cleanup() error {
 	return nil
 }
 
+// cleanupReset removes all previous batches.
+// Must be called under lock.
+func (s *store) CleanupReset() error {
+
+	var evictions []*postage.Batch
+
+	err := s.store.Iterate(valueKeyPrefix, func(key, value []byte) (stop bool, err error) {
+
+		b, err := s.get(valueKeyToID(key))
+		if err != nil {
+			return false, err
+		}
+
+		evictions = append(evictions, b)
+
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, b := range evictions {
+
+		err := s.evictFn(b.ID)
```

Comment:
we might as well nuke the whole localstore in one call, instead of evicting content one batch at a time

Reply:
What about nodes that have substantial numbers of chunks explicitly pinned?  IMHO, nuking the whole localstore would be a severe dis-service to anyone expecting to actually retain data locally!